### ISSUE TITLE
Refactor contribution function

### ISSUE THREAD

**Issue by wtbarnes** [chianti-atomic/ChiantiPy#129]
Instead of bringing up several dialogs and plots, `ion.gofnt()` should just return to the user (or set the `ion.Gofnt` dictionary) a N(temperature/density) by N(wavelength) array with the associated contribution function values. The user can then decide which transitions are the important ones. Since the wavelength values are returned to the user in the `ion.Gofnt` dictionary anyway, it should still be easy to select particular transitions based on their wavelength values.

**Comment by kdere** [comment]
As it is, is is useful for just quickly trying to find what temperature produces the maximum Gofnt for a line or 2 that you know.
When I try to interpret a set of line intensities, I associate the observed intensities with the ion.Intensity dict by way of closest wavelength.  However, how to do this is not obvious to the user.  I need to see exactly what the user is getting in .Intensity vs Gofnt  

So, I would say keep gofnt as it is, or rename it, and find a way to either document how to associate contribution function with a given set of lines.

**Comment by wtbarnes** [comment]
In general, I think the focus should be on returning as much information as possible to the user. As it stands now, the user is only returned a small number of lines and the need to pick lines in the GUI (or CLI) and then plot the gofnt makes non-interactive use of this method difficult.

Provided the user is returned the emissivity array and the associated wavelengths, it should be very simple (e.g. using `np.sort` and/or `np.where`) to find the top lines. We could include an example of this in the docstring.

In summary, three suggestions for overall improvement of this method:

1. Remove wavelength range selection. This does not save time and again can easily be done by the user.
2. Remove all plotting and GUI/CLI selections
3. Add example in the docstring for selecting top lines based on emissivity values 

I'm willing to take care of these changes.

**Comment by kdere** [comment]
I think we should add an additonal method such as gofntAll or something like that that does what you say.  I would like to keep the current gofnt, we could rename it something like gofntInteract but I find it very useful.

**Comment by wtbarnes** [comment]
I wonder if it might make more sense to place all of the interactive functionality in `ion` in a separate class, e.g. `ionInteract` (or something like that). 

This would mean removing all plotting commands and anything that involves a selection dialog.

**Comment by kdere** [comment]
I would prefer to keep it all in ion.  There really are not all that many interactive methods and I would rather just have one main import.

**Comment by kdere** [comment]
I think  the .intensity method provides everything you need for the gofnt of all or a set of lines from a given ion.  It is just not called that.
